src/EditorFeatures/Core/CommandHandlers/GoToNextAndPreviousMethodCommandHandler.cs

```diff
@@ -0,0 +1,185 @@
+            if (syntaxFactsService == null)
+            {
+                return null;
+            }
+
+            var root = document.GetSyntaxRootAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var members = syntaxFactsService.GetMethodLevelMembers(root);
+
+            if (members.Count == 0)
+            {
+                return null;
+            }
+
+            var currentMember = GetMember(syntaxFactsService, caretPosition, root);
+            int indexOfCurrentMember;
+            if (currentMember == null)
+            {
+                // We're not contained in any member currently.  Find the first member to start after our
+                // position and use that.
+                for (indexOfCurrentMember = 0; indexOfCurrentMember < members.Count; indexOfCurrentMember++)
+                {
+                    if (members[indexOfCurrentMember].FullSpan.Start > caretPosition)
+                    {
+                        break;
+                    }
+                }
+
+                if (indexOfCurrentMember == members.Count)
+                {
+                    // After the last member, count ourselves as part of that last member, so that we wrap
+                    // to the first, or back to the beginning of last.
+                    indexOfCurrentMember--;
+                }
+
+                currentMember = members[indexOfCurrentMember];
+            }
+            else
+            {
+                indexOfCurrentMember = members.IndexOf(currentMember);
+                if (indexOfCurrentMember < 0)
+                {
+                    Debug.Fail("Couldn't find current member in members?");
+                    return null;
+                }
+            }
+
+            if (next)
+            {
+                // If we're in leading trivia, we just want to go to the start of this member.
+                if (caretPosition >= currentMember.Span.Start)
+                {
+                    indexOfCurrentMember++;
+                    if (indexOfCurrentMember == members.Count)
+                    {
+                        indexOfCurrentMember = 0;
+                    }
+                }
+            }
+            else
+            {
+                // If we're in trailing trivia, we just want to go to the start of this member.
+                if (caretPosition <= currentMember.Span.End)
+                {
+                    indexOfCurrentMember--;
+                    if (indexOfCurrentMember < 0)
+                    {
+                        indexOfCurrentMember = members.Count - 1;
+                    }
+                }
+            }
+
+            // TODO: Better position within the node (e.g. attributes?)
+            return members[indexOfCurrentMember].Span.Start;
+        }
+
+        private static SyntaxNode GetMember(ISyntaxFactsService syntaxFactService, int position, SyntaxNode root)
```

Comment:
I had this first, when I was being lazy and not making it work when you're outside a type, etc.  Take a look at the latest version which simplifies all of this with `Array.BinarySearch`.